modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala

```diff
@@ -1,32 +1,65 @@
 package org.tessellation.sdk.domain.healthcheck.services
 
-import cats.effect.Ref
+import cats.Applicative
+import cats.effect.{Ref, Spawn}
 import cats.syntax.bifunctor._
 import cats.syntax.flatMap._
 import cats.syntax.functor._
 import cats.syntax.traverse._
-import cats.{Applicative, Monad}
 
-import org.tessellation.sdk.domain.healthcheck.types.ConsensusRounds
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.sdk.domain.healthcheck.types._
 
-abstract class HealthCheckConsensus[F[_]: Monad]() extends HealthCheck[F] {
+abstract class HealthCheckConsensus[F[_]: Spawn]() extends HealthCheck[F] {
   def allRounds: Ref[F, ConsensusRounds[F]]
   def roundsInProgress = allRounds.get.map(_.inProgress)
+  def historicalRounds = allRounds.get.map(_.historical)
+
+  def peersUnderConsensus: F[Set[PeerId]] =
+    roundsInProgress.map(_.keySet.map(_.id))
 
   final override def trigger(): F[Unit] =
     triggerRound()
 
-  def triggerRound(): F[Unit] =
+  private def triggerRound(): F[Unit] =
     roundsInProgress.flatMap { inProgress =>
       if (inProgress.nonEmpty) manageRounds(inProgress) else Applicative[F].unit
     }
 
-  def manageRounds(rounds: ConsensusRounds.InProgress[F]): F[Unit] =
+  private def manageRounds(rounds: ConsensusRounds.InProgress[F]): F[Unit] = {
```

Comment:
Don't you want to import `InProgress` and `Finished` directly?

Reply:
No, I think it looks cleaner that way